### DISCUSSION TITLE
exec /docker-entrypoint.sh: exec format error

### DISCUSSION THREAD

**Discussion by githubsharezone** [postalserver/postal#2338] [Help with using Postal]
when execute the command "postal initialize" after the execution show error " exec /docker-entrypoint.sh: exec format error " and when i execute root@armdkr02:/opt/postal/config# postal make-user root
exec /docker-entrypoint.sh: exec format error

root@armdkr02:/opt/postal/config# docker -v
Docker version 20.10.21, build baeda1f
root@armdkr02:/opt/postal/config#

root@armdkr02:/opt/postal/config# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
root@armdkr02:/opt/postal/config#

**Comment by willpower232** [comment]
Apparently this means the top of the script file is incorrect somehow.

Are you able to run `postal dc web head /docker-entrypoint.sh` and see what output you get?

**Reply by githubsharezone** [reply]
root@armdkr02:~# postal dc web head /docker-entrypoint.sh

Usage:  docker compose [OPTIONS] COMMAND

Docker Compose

Options:
      --ansi string                Control when to print ANSI control characters
                                   ("never"|"always"|"auto") (default "auto")
      --compatibility              Run compose in backward compatibility mode
      --env-file string            Specify an alternate environment file.
  -f, --file stringArray           Compose configuration files
      --profile stringArray        Specify a profile to enable
      --project-directory string   Specify an alternate working directory
                                   (default: the path of the Compose file)
  -p, --project-name string        Project name

Commands:
  build       Build or rebuild services
  convert     Converts the compose file to platform's canonical format
  cp          Copy files/folders between a service container and the local filesystem
  create      Creates containers for

**Reply by willpower232** [reply]
my mistake, I always forget the important bit `postal dc exec web head /docker-entrypoint.sh`

**Reply by githubsharezone** [reply]
i execute the commamd postal dc exec web head /docker-entrypoint.sh

but return 

service "web" is not running container #1

the docker runnin are 

7f88d2a13230   rabbitmq:3.8   "docker-entrypoint.s…"   4 days ago   Up 4 days   4369/tcp, 5671/tcp, 15691-15692/tcp, 25672/tcp, 127.0.0.1:5672->5672/tcp   postal-rabbitmq
1f73ef90e543   mariadb        "docker-entrypoint.s…"   4 days ago   Up 4 days   127.0.0.1:3306->3306/tcp                                                   postal-mariadb
root@armdkr02:~#

**Reply by willpower232** [reply]
ah so none of Postal is running, does the `arm` in `armdkr02` mean you're running an ARM server? I don't think that will work as the docker images are not arm based which may explain the error you're seeing.

**Reply by githubsharezone** [reply]
Is an esxi instance on raspberry . 

but running 

PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
root@armdkr02:~#

Architecture:        aarch64
Byte Order:          Little Endian
CPU(s):              2
On-line CPU(s) list: 0,1
Thread(s) per core:  1
Core(s) per socket:  2
Socket(s):           1
NUMA node(s):        1
Vendor ID:           ARM
Model:               3
Model name:          Cortex-A72
Stepping:            r0p3
BogoMIPS:            108.00
NUMA node0 CPU(s):   0,1
Flags:               fp asimd evtstrm crc32 cpuid

**Reply by willpower232** [reply]
As I said, that would explain it. There have been a few people trying to make Postal work on arm-based servers but I don't think any have succeeded yet, you would probably have to make your own Docker image.

**Reply by githubsharezone** [reply]
Too bad we can't run postal on arm . :-(
but even without container it is not possible to make it work ? 
Example from scratch installation ?

**Reply by willpower232** [reply]
its not impossible but you would have to know enough about running processes from ruby on rails to run all the necessary parts, the code that originally did that was probably removed as part of the v2 upgrade
